apis/datacatalog/v1alpha1/entry_identity.go

```diff
@@ -61,10 +61,21 @@ func (p *EntryParent) String() string {
 // NewEntryIdentity builds a EntryIdentity from the Config Connector DataCatalogEntry object.
 func NewEntryIdentity(ctx context.Context, reader client.Reader, obj *DataCatalogEntry) (*EntryIdentity, error) {
 	// --- Determine Parent ---
-	if obj.Spec.EntryGroupRef == nil || obj.Spec.EntryGroupRef.External == "" {
+	if obj.Spec.EntryGroupRef == nil || (obj.Spec.EntryGroupRef.External == "" && obj.Spec.EntryGroupRef.Name == "") {
```

Comment:
Maybe I missed it in earlier PR reviews, but this looks unnecessary. The check should be already included in `obj.Spec.EntryGroupRef.NormalizedExternal`?